Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/dialogController/UserMenuController.java

```diff
@@ -213,17 +213,19 @@ public void onEvent(Event event) {
                         try {
                             boolean logoutSuccess = false;
 
+                            final Session session = Sessions.getCurrent();
+
                             if (config.isUseKeycloakSso()) {
-                                logoutSuccess =
-                                        managerService.logoutUserAllSessions(
-                                                currentUser.getUsername(),
-                                                config.getSecurityMsHttpLogoutUrl(),
-                                                config.getSecurityMsHttpsLogoutUrl());
-                                LOGGER.info("\nlogoutSuccess: {}", logoutSuccess);
+                                if (event.getData().equals(session)) {
+                                    logoutSuccess =
+                                            managerService.logoutUserAllSessions(
+                                                    currentUser.getUsername(),
+                                                    config.getSecurityMsHttpLogoutUrl(),
+                                                    config.getSecurityMsHttpsLogoutUrl());
+                                    LOGGER.info("\nlogoutSuccess: {}", logoutSuccess);
+                                }
                             }
 
-                            final Session session = Sessions.getCurrent();
-
                             if ((session == null || event.getData().equals(session)) || (logoutSuccess)) {
                                 Clients.evalJavaScript("window.close()");
                                 Executions.sendRedirect("/j_spring_security_logout");
```

Comment:
The same comment as discussed before: this event listener is visible to all sessions, the keycloak logout doesn't check whether the received event is for this session (i.e. whether it is targeted this currentUser or not) while the redirect to Spring logout checks that with "event.getData).equals(session)".